pandas/core/internals/blocks.py

```diff
@@ -1809,6 +1809,11 @@ def _slice(self, slicer):
             if not com.is_null_slice(slicer[0]):
                 raise AssertionError("invalid slicing for a 1-ndim categorical")
             slicer = slicer[1]
+        elif not isinstance(slicer, tuple) and self.ndim == 2:
+            # reached via getitem_block via _slice_take_blocks_ax0
+            # TODO(EA2D): wont be necessary with 2D EAs
+            # treat this like (slicer, slice(None)
+            slicer = slice(None)
```

Comment:
What about slices that wouldn't include this column? Eg slice(1, 2, None) shouldn't just be converted to slice(None) ?

Reply:
_slice_take_blocks_ax0 shouldnt call this in that case

Reply:
But it seems it does though. If I put a breakpoint here, it halts with `df.iloc[:, 1:2]`

Reply:
comments addressed, i think

Reply:
Can you add this as a test case?